cpp/src/arrow/array/array_primitive.h

```diff
@@ -108,11 +108,11 @@ class ARROW_EXPORT DayTimeIntervalArray : public PrimitiveArray {
   // For compatibility with Take kernel.
   TypeClass::DayMilliseconds GetView(int64_t i) const { return GetValue(i); }
 
-  int32_t byte_width() const { return sizeof(TypeClass::DayMilliseconds); }
-
   const uint8_t* raw_values() const { return raw_values_ + data_->offset * byte_width(); }
 
  protected:
+  static constexpr int32_t byte_width() { return sizeof(TypeClass::DayMilliseconds); }
```

Comment:
Not sure why you made this protected? The byte width may be useful.

Reply:
It's not used anywhere except in this class. If needed it can easily be reintroduced but it seemed better to minimize the public API

Reply:
Well, it was public so could be used by anybody really.

Reply:
I also don't think this change belongs in this PR